Sources/Basic/FileSystem.swift

```diff
@@ -351,6 +365,24 @@ private class LocalFileSystem: FileSystem {
             break
         }
     }
+    
+    func writeFileContents(_ path: AbsolutePath, bytes: ByteString, atomically: Bool) throws {
+        // Perform non-atomic writes using the fast path
+        guard atomically else {
+            return try writeFileContents(path, bytes: bytes)
+        }
+        let temp = try TemporaryFile(deleteOnClose: false)
```

Comment:
We should create the temporary file in the directory where the actual file needs to be written as it could be expensive to move files from OS's temp directory to final location (which could be a network drive).